src/services/auth-service.ts

```diff
@@ -0,0 +1,128 @@
+import { repository } from '@loopback/repository';
+import { JWTService } from './jwt-service';
+import { securityId, UserProfile } from '@loopback/security';
+import { TokenService } from '@loopback/authentication';
+import { TokenServiceBindings } from '@loopback/authentication-jwt';
+import { inject } from '@loopback/core';
+import winston from 'winston';
+import { LogConfig } from '../config/logConfig';
+import { HttpErrors } from '@loopback/rest';
+import { TokenKvRepository } from '../repositories';
+import { TokenKv } from '../models';
+
+
+
+export interface Tokens {
+  accessToken: string,
+  refreshToken: string,
+  expiresIn: number
+}
+
+export class AuthService {
+  constructor(
+    @repository(TokenKvRepository)
+    public redisRepo: TokenKvRepository,
+    @inject(TokenServiceBindings.TOKEN_SERVICE)
+    public jwtService: TokenService,
+    public logger = winston.loggers.get(LogConfig.logName),
+  ) { }
+
+  /**
+   * Takes a users and returns access and refresh tokens. 
+   * It also saves tokens into redis with ttl of 2 hours
+   * @param userP User Profile
+   * @returns 
+   */
+  async createTokens(userP: UserProfile): Promise<Tokens> {
+    let accToken = '';
+    let refToken = '';
+
+    const userId = String(userP[securityId])
+
+    try {
+      accToken = await this.jwtService.generateToken(userP)
+      refToken = await JWTService.prototype.generateRefreshToken(userP)
+
+
+      // Save tokens in redis
+      if (accToken && refToken) {
```

Comment:
is it necessary to have the existence check in the `if`?
I think an error will be thrown if there is any token generation failure and the `if` won't be reached at all, no?

Reply:
Yes, `if` won't be reached when token generation fails, you are correct. I think it would be better to check for the  existence of previous tokens and removing them before pushing the new tokens. 
This way, if there is an error generating the token, the previous token will still be available in Redis until it expires.